File: tools4j-lang/src/main/java/com/github/panpf/tools4j/lang/Objectx.java
/*
 * Copyright (C) 2020 panpf <panpfpanpf@outlook.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.panpf.tools4j.lang;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Object tool method
 */
public class Objectx {

    /**
     * Returns a simple string representation of the object.
     * value of:
     * <blockquote>
     * <pre>
     * getClass().getSimpleName() + '@' + Integer.toHexString(hashCode())
     * </pre></blockquote>
     *
     * @return a string representation of the object.
     */
    @NotNull
    public static String toSimpleString(@Nullable Object object) {
        return object != null ? object.getClass().getSimpleName() + "@" + Integer.toHexString(object.hashCode()) : "null";
    }
}

File: tools4j-lang-ktx/src/main/java/com/github/panpf/tools4j/lang/ktx/Objectx.kt
@file:Suppress("NOTHING_TO_INLINE")

package com.github.panpf.tools4j.lang.ktx

import com.github.panpf.tools4j.lang.Objectx

/*
 * Object tool method
 */

/**
 * Returns a simple string representation of the object.
 * value of:
 * <blockquote>
 * <pre>
 * getClass().getSimpleName() + '@' + Integer.toHexString(hashCode())
 * </pre></blockquote>
 *
 * @return  a string representation of the object.
 */
inline fun Any?.toSimpleString(): String = Objectx.toSimpleString(this)